ow_faults_injection/src/FaultInjector.cpp

```diff
@@ -0,0 +1,199 @@
+    output.position[index]  = FAULT_ZERO_TELEMETRY;
+    flag_msg.flags[index] = SET_FLAG;
+  }
+  if (m_faults.prox_pitch_effort_failure && findJointIndex(J_PROX_PITCH, index)) {
+    output.effort[index]  = FAULT_ZERO_TELEMETRY;
+    flag_msg.flags[index] = SET_FLAG;
+  }
+
+  if (m_faults.dist_pitch_encoder_failure && findJointIndex(J_DIST_PITCH, index)) {
+    output.position[index]  = FAULT_ZERO_TELEMETRY;
+    flag_msg.flags[index] = SET_FLAG;
+  }
+  if (m_faults.dist_pitch_effort_failure && findJointIndex(J_DIST_PITCH, index)) {
+    output.effort[index]  = FAULT_ZERO_TELEMETRY;
+    flag_msg.flags[index] = SET_FLAG;
+  }
+
+  if (m_faults.hand_yaw_encoder_failure && findJointIndex(J_HAND_YAW, index)) {
+    output.position[index]  = FAULT_ZERO_TELEMETRY;
+    flag_msg.flags[index] = SET_FLAG;
+  }
+  if (m_faults.hand_yaw_effort_failure && findJointIndex(J_HAND_YAW, index)) {
+    output.effort[index]  = FAULT_ZERO_TELEMETRY;
+    flag_msg.flags[index] = SET_FLAG;
+  }
+
+  if (m_faults.scoop_yaw_encoder_failure && findJointIndex(J_SCOOP_YAW, index)) {
+    output.position[index]  = FAULT_ZERO_TELEMETRY;
+    flag_msg.flags[index] = SET_FLAG;
+  }
+  if (m_faults.scoop_yaw_effort_failure && findJointIndex(J_SCOOP_YAW, index)) {
+    output.effort[index]  = FAULT_ZERO_TELEMETRY;
+    flag_msg.flags[index] = SET_FLAG;
+  }
+
+  m_joint_state_remapped_pub.publish(output);
+  m_joint_state_flags_pub.publish(flag_msg);
+}
+
+void FaultInjector::distPitchFtSensorCb(const geometry_msgs::WrenchStamped& msg)
+{
+  if (!m_faults.groups.ft_sensor_faults.enable) {
+    m_dist_pitch_ft_sensor_pub.publish(msg);
+    return;
+  }
+
+  auto out_msg = msg;
+
+  if (m_faults.groups.ft_sensor_faults.zero_signal_failure) {
+    out_msg.wrench.force = geometry_msgs::Vector3();
+    out_msg.wrench.torque = geometry_msgs::Vector3();
+  }
+
+  auto mean = m_faults.groups.ft_sensor_faults.signal_bias_failure;
+  auto stddev = m_faults.groups.ft_sensor_faults.signal_noise_failure;
+  // TODO: consider optimizing this by re-creating the distribution only when
+  // mean and stddev values change
+  auto normal_dist = std::normal_distribution<float>(mean, stddev);
+  out_msg.wrench.force.x += normal_dist(m_random_generator);
+  out_msg.wrench.force.y += normal_dist(m_random_generator);
+  out_msg.wrench.force.z += normal_dist(m_random_generator);
+  out_msg.wrench.torque.x += normal_dist(m_random_generator);
+  out_msg.wrench.torque.y += normal_dist(m_random_generator);
+  out_msg.wrench.torque.z += normal_dist(m_random_generator);
+
+  m_dist_pitch_ft_sensor_pub.publish(out_msg);
+}
+
+void FaultInjector::checkCamFaults(){
+  m_cam_fault = m_faults.camera_left_trigger_failure;
+}
+
+template<typename group_t, typename item_t>
+int FaultInjector::findPositionInGroup(const group_t& group, const item_t& item)
+{
+  auto iter = std::find(group.begin(), group.end(), item);
+  if (iter == group.end())
+    return -1;
```

Comment:
Bracket this, and the similar return in the next function, for more clarity and safety.  Optionally, here's a slick way to code the last two statements: `return (iter == group.end()) ? -1 : group.begin();`